hs_file_types/models/geofeature.py

```diff
@@ -249,125 +249,69 @@ def check_files_for_aggregation_type(cls, files):
-            aggregation_file_count=len(shape_files))
-
-        msg = "GeoFeature aggregation. Error when creating aggregation. Error:{}"
-        with transaction.atomic():
-            # create a GeoFeature logical file object to be associated with
-            # resource files
-            logical_file = cls.initialize(base_file_name, resource)
-            try:
-                if folder_path is None:
-                    # we are here means aggregation is being created by selecting a file
-
-                    if create_new_folder:
-                        # create a folder for the geofeature file type using the base file
-                        # name as the name for the new folder
-                        upload_folder = cls._create_aggregation_folder(resource, file_folder,
-                                                                       base_file_name)
-
-                        log.info("Folder created:{}".format(upload_folder))
-                        aggregation_folder_created = True
-                        # create logical file record in DB
-                        logical_file.save()
-                        if res_file.extension.lower() == ".shp":
-                            # selected file is a shp file - that means we didn't create any new
-                            # files - we just need to copy all the existing files to a new folder
-                            # and makes those copied files part of the new aggregation
-                            tgt_folder = upload_folder
-                            logical_file.copy_resource_files(resource, shp_res_files, tgt_folder)
-                            res_files_to_delete.extend(shp_res_files)
+        with FileTypeContext(aggr_cls=cls, user=user, resource=resource, file_id=file_id,
+                             folder_path=folder_path,
+                             post_aggr_signal=post_add_geofeature_aggregation,
+                             is_temp_file=True) as ft_ctx:
 
-                        else:
-                            # selected file must be a zip file - add the extracted files to
-                            # the resource and make those files part of the aggregation
-                            logical_file.add_files_to_resource(
-                                resource=resource, files_to_add=shape_files,
-                                upload_folder=upload_folder)
-                            res_files_to_delete.append(res_file)
+            res_file = ft_ctx.res_file
+            try:
+                meta_dict, shape_files, shp_res_files = extract_metadata_and_files(resource,
+                                                                                   res_file)
+            except ValidationError as ex:
+                log.exception(ex.message)
+                raise ex
+
+            file_name = res_file.file_name
+            # file name without the extension
+            base_file_name = file_name[:-len(res_file.extension)]
```

Comment:
noticed line 270 will generate base_file_name to end with ```.```, right? Should this trailing dot be filtered out from base_file_name as well?

Reply:
```res_file.extension``` has the starting char as ```.```  (e.g,  ```.txt```) so it is already removed as part of the slicing operation.